FBReader/src/main/java/org/geometerplus/android/fbreader/benetech/AddToReadingListDialogActivity.java

```diff
@@ -73,10 +106,47 @@ public void onClick(View view) {
     }
 
     private void addToReadingList() {
-        ReadingList readingList = readingListsItems.get(selectedReadingListIndex).readingList;
+        new AddTitleToReadingListTask().execute();
+    }
 
+    class AddTitleToReadingListTask extends AsyncTask<String, Void, Boolean> {
+        ReadingList readingList;
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            readingList = readingListsItems.get(selectedReadingListIndex).readingList;
+        }
+
+        protected Boolean doInBackground(String... urls) {
+            Object result = null;
+            try {
+                SharedPreferences login_preference = PreferenceManager.getDefaultSharedPreferences(AddToReadingListDialogActivity.this);
+                String username = login_preference.getString(Bookshare_Webservice_Login.USER, "");
+                String password = login_preference.getString(Bookshare_Webservice_Login.PASSWORD, "");
+
+                BookshareHttpOauth2Client client =  new BookshareHttpOauth2Client();
+                HttpsURLConnection urlConnection = client.createBookshareApiUrlConnection(username, password);
+
+                String response = client.requestData(urlConnection);
+                JSONObject jsonResponse = new JSONObject(response);
+                String accessToken = jsonResponse.getString(BookshareHttpOauth2Client.ACCESS_TOKEN_CODE);
+
+                result  = client.postTitleToReadingList(accessToken, readingList.getBookshareId(), bookId);
+            } catch (Exception e) {
+                Log.e(getClass().getSimpleName(), e.getMessage(), e);
+                result = new JSONArray();
```

Comment:
this feels odd since it will cause the method to return true if there was an exception.